### PR TITLE
decorate a class

### PR DESCRIPTION
**Pull request by mohamedelshamy55**
In this project, you will use the Decorator design pattern to validate and correct the names of people.
Implement composition (as an example of the design pattern).
Run a program using the command line.

### CODE REVIEW

File: crorrector.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+class Corrector
+  def correct_name(name)
+    name.capitalize
+    name[0..10]
+  end
```
Comment (TedLivist) [inline crorrector.rb]: - [x] Your method does not return a capitalized string. Kindly fix that. You can resolve it by adding a bang sign (`!`) to the `capitalize` method.
- [x] Your method is also supposed to return only 10 character string but it returns 11. Kindly fix this. `[0..9]` or `[0...10]` should work.

  elshamy-corr

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @mohamedelshamy55,

Good job so far!
✔️ Implementation of Corrector instance in Person class
✔️ Implementation of `validate_name` method.
However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
#### 🛑 🛑 Check this section as well 🛑 🛑
- [ ] **In your README, kindly add instructions on how to get your app locally**
- [ ] _Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by BrianSammit — APPROVED** [review]
# STATUS: APPROVED :green_circle: :clap: 

Hello @mohamedelshamy55, 
Your project is complete!  There is nothing else to say other than... Great job 👏 
Congratulations! 🎉 

TO HIGHLIGHT:

- Your project is professional 
- Your repo is professional 
- Well structured files 


Now you can merge this branch and move on :1st_place_medal: 
HAPPY CODING AND GOOD LUCK WITH THE COMMING ONES :crossed_fingers: :four_leaf_clover: 

well-1

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: crorrector.rb (hunk 1 above)
Comment (mohamedelshamy55) [reply crorrector.rb]: thank you

**Comment by mohamedelshamy55** [comment]
@TedLivist @BrianSammit Thank you for finding time to review my project. I appreciate.

**Pull request merged**